src/pygama/raw/buffer_processor/buffer_processor.py

```diff
@@ -0,0 +1,345 @@
+from __future__ import annotations
+
+import copy
+import logging
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+import pygama.lgdo as lgdo
+from pygama.dsp.processing_chain import build_processing_chain as bpc
+from pygama.lgdo import Array, ArrayOfEqualSizedArrays
+
+if TYPE_CHECKING:
+    from pygama.raw.raw_buffer import RawBuffer
+
+log = logging.getLogger(__name__)
+
+
+def buffer_processor(rb: RawBuffer) -> None:
+    """
+    Takes in a :class:`.RawBuffer`, performs any of the four processes specified from the :class:`.RawBuffer`'s ``proc_spec`` attribute:
+    - windows objects with a name specified by the first argument passed in the ``proc_spec``, the window start and stop indices are the next two
+    arguments, and then updates the rb.lgdo with a name specified by the last argument. If the object is an :func:`pygama.lgdo.WaveformTable`, then
+    the ``t0`` and ``dt`` attributes are updated accordingly.
+    - performs DSP given by the "dsp_config" key in the ``proc_spec``. See :module:`pygama.dsp` for more information on DSP config dictionaries.
+    All fields in the output of the DSP are written to the rb.lgdo
+    - drops any requested fields from the rb.lgdo
+    - updates the data types of any field with its requested datatype in ``proc_spec``
+
```

Comment:
Make this docstring read more like an API for the proc_spec. Something like:

Currently implemented attributes:
- "window": [in_name, start_index, stop_index, out_name]
  Windows objects with a name...